### ISSUE TITLE
approximate_predict generates new clusters when using cluster_selection_epsilon

### ISSUE THREAD

**Issue by mczerny** [scikit-learn-contrib/hdbscan#375]
I am clustering geolocation data, and I noticed that when I try to predict clusters on new data I get more clusters than I saw in the original training data.

Here is how I fit the data:
```
hdb = hdbscan.HDBSCAN(min_cluster_size=2, min_samples=1, metric='haversine', 
                      cluster_selection_epsilon=epsilon, cluster_selection_method='eom', 
                      prediction_data=True).fit(radian_locs)
```

This gives me 45 clusters. However, when I call:
`pred_clusters, conf = hdbscan.approximate_predict(hdb, test_radian_locs)`
I get a total of 232 clusters. **I noticed that if I remove the `cluster_selection_epsilon` argument, the number of clusters match.** Is there a way to preserve this in the `approximate_predict` function?

**Comment by lmcinnes** [comment]
It is quite possible that these two features don't actually interact properly. I believe the approximate_predict is just using the clusters that would have been selected with the ``cluster_selection_epsilon`` argument, and thus produces different clusters.

**Comment by mczerny** [comment]
Looking through the source code it seems to be that way; `approximate_predict` doesn't use the `get_clusters` method in `_hdbscan_tree.pyx`, which is what seems to do the cluster merging. Do you think it would be straightforward/useful to implement the same behavior for `approximate_predict`? I think for now I can try a simple KNN classifier to predict new cluster labels.

**Comment by lmcinnes** [comment]
It may be a little tricky, but is probably feasible. I think for now the KNN classifier is more likely to the the useful approach. Sorry.

**Comment by apcamargo** [comment]
Just for future reference in case someone decides to tackle this issue: https://github.com/scikit-learn-contrib/hdbscan/issues/360 seems to be related to this bug.

**Comment by sabarish-akridata** [comment]
Have a look at [PR #398 ](https://github.com/scikit-learn-contrib/hdbscan/pull/398) for one solution to this issue.

**Comment by zonnalobo** [comment]
I'm also fasing the same issue. I use several number of `cluster_selection_epsilon` in `approximate_predict` but the teh prediction results provide similar results for all number as we can see below:

![image](https://user-images.githubusercontent.com/42733716/161191475-34602024-73ab-4574-be7e-c5245c469fcc.png)

**Comment by sarahshi** [comment]
I am seeing the same issue at the moment. My latent space is currently composed of dens clusters with irregular shapes, and I've found that HDBSCAN clusters the data well when I apply the cluster_selection_epsilon. See attached for an example of this, with my test/train data on the left and the new data on the right. Most values are predicted to belong to these errant value clusters. 
![cluster_selection_epsilon](https://user-images.githubusercontent.com/12854070/231280734-b79e51f8-f369-48a8-a964-e52ffd497900.png)

See the predicted labels for the test/train data (left) and new data (right) when I remove the cluster_selection_epsilon parameter. I've done quite a bit of hyperparameter tuning but can't seem to improve things too much more. I would greatly appreciate any advice for better approaches to this problem. 
![no_cluster_selection_epsilon](https://user-images.githubusercontent.com/12854070/231280886-1798834e-1ddb-4971-b246-d31e903c57d7.png)

**Comment by sarahshi** [comment]
#398 solved this issue with the ability to select for cluster_selection_epsilon. I had to filter out all the -1 predictions, which solved this problem. See attached for updated clustering with HDBSCAN_flat and predictions from approximate_predict_flat (with the same cluster_selection_epsilon value). 

![output](https://user-images.githubusercontent.com/12854070/231812740-2ddc7060-740b-4d75-8623-35e3ee61beb4.png)
